File: artifacts/libzahl-1.0/decompose/src/zbtest.c
int zbtest(z_t a, size_t bit)
{
  size_t chars;
  if (zzero(a))
  {
    return 0;
  }
  chars = bit >> 5;
  if (chars >= a->used)
  {
    return 0;
  }
  bit = bit & (32 - 1);
  return (a->chars[chars] >> bit) & 1;
}


/*** DEPENDENCIES:
inline static int zzero(z_t a)
{
  return !a->sign;
}


----------------------------
typedef struct 
{
  int sign;
  size_t used;
  size_t alloced;
  zahl_char_t *chars;
} z_t[1]
----------------------------
***/

File: artifacts/libzahl-1.0/manual_translation/src/zbtest.rs
use crate::*;

pub fn zbtest(a: &z_t, bit: usize) -> i32 {
    let chars;
    if zzero(a) {
        return 0;
    }
    chars = bit >> 5;
    if chars >= a[0].used {
        return 0;
    }
    let bit = bit & (32 - 1);
    match &a[0].chars {
        Some(chars_array) => ((chars_array[chars] >> bit) & 1) as i32,
        None => 0,
    }
}
